src/libraries/System.Runtime/tests/System/StringTests.cs

```diff
@@ -546,6 +546,7 @@ public static void Join_Char_StringArray(char separator, string[] values, int st
             {
                 Assert.Equal(expected, string.Join(separator, values));
                 Assert.Equal(expected, string.Join(separator, (IEnumerable<string>)values));
+                Assert.Equal(expected, string.Join(separator, values.Concat(new string[0])));
```

Comment:
Consider adding a comment to this line stating that it's intended to exercise fallback (non-`List<T>`) code paths. Otherwise it's difficult to ascertain what this line is trying to do.